test/integration/network/services_test.go

```diff
@@ -0,0 +1,455 @@
+		},
+		{
+			name:            "DualStack enabled and IPv6 IPv4",
+			serviceCIDR:     "2001:db8:1::/48,10.0.0.0/16",
+			defaultIPFamily: v1.IPv6Protocol,
+			dualStack:       true,
+		},
+		{
+			name:            "DualStack enabled and IPv6 IPv4 IPv6",
+			serviceCIDR:     "2001:db8:1::/48,10.0.0.0/16,fd00:192:168::/48",
+			defaultIPFamily: v1.IPv6Protocol,
+			dualStack:       true,
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			etcd := framework.SharedEtcd()
+			// cleanup the registry storage
+			defer registry.CleanupStorage()
+			// start a kube-apiserver
+			server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{
+				"--service-cluster-ip-range", tc.serviceCIDR,
+				"--advertise-address", "10.0.0.1",
+				"--feature-gates", fmt.Sprintf("IPv6DualStack=%t", tc.dualStack),
+			}, etcd)
+			// create a client
+			client, err := kubernetes.NewForConfig(server.ClientConfig)
+			if err != nil {
+				t.Errorf("error creating client: %v", err)
+			}
+
+			// verify client is working
+			if err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+				_, err = client.CoreV1().Endpoints("default").Get("kubernetes", metav1.GetOptions{})
+				if err != nil {
+					t.Logf("error fetching endpoints: %v", err)
+					return false, nil
+				}
+				return true, nil
+			}); err != nil {
+				t.Errorf("server without enabled endpoints failed to register: %v", err)
+			}
+
+			// Create a service ClusterIP without IPFamily so it's created with the default IP family
+			svc := createService("svc-transition", nil, v1.ServiceTypeClusterIP)
+			if _, err := client.CoreV1().Services(metav1.NamespaceDefault).Create(svc); err != nil {
+				t.Errorf("unexpected error text: %v", err)
+			}
+
+			// Transition the service over all the types of services and check that the IPFamily is inmutable
+			ipFamily := tc.defaultIPFamily
+
+			svcType := []v1.ServiceType{
+				v1.ServiceTypeExternalName,
+				v1.ServiceTypeNodePort,
+				v1.ServiceTypeLoadBalancer,
+				v1.ServiceTypeClusterIP,
+				v1.ServiceTypeClusterIP, // no-op
+			}
+			for _, serviceType := range svcType {
+				// obtain the services and check the Spec.IPFamily is correct
+				s, err := client.CoreV1().Services(metav1.NamespaceDefault).Get(svc.ObjectMeta.Name, metav1.GetOptions{})
+				if err != nil {
+					t.Fatalf("unexpected error getting the service: %v", err)
+				}
+				// single family Spec.IPFamily must be nil
+				if !tc.dualStack {
+					if s.Spec.IPFamily != nil {
+						t.Fatalf("expected service IP family nil: %s ", s.Spec.ClusterIP)
+					}
+				} else if err = validateServiceAndClusterIPFamily(s, ipFamily); err != nil {
+					t.Fatalf("unexpected service IP family %s: %v", s.Spec.ClusterIP, err)
+				}
+
+				resourceVersion, err := strconv.ParseFloat(s.ResourceVersion, 64)
+				if err != nil {
+					t.Errorf("error converting resource version to int: %v", err)
+				}
+				resourceVersion++
```

Comment:
why are you incrementing the resourceVersion? the update should be sent with the same version you received from the Get

Reply:
I did that and it throwed this error ```metadata.resourceVersion: Invalid value: "": must be specified for an update``` and this was the answer I've found googling around.
Maybe the error was related to the other fields? let me try again